jest/src/test/java/io/searchbox/cluster/RerouteIntegrationTest.java

```diff
@@ -0,0 +1,148 @@
+package io.searchbox.cluster;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.searchbox.client.JestResult;
+import io.searchbox.cluster.reroute.RerouteAllocate;
+import io.searchbox.cluster.reroute.RerouteCancel;
+import io.searchbox.cluster.reroute.RerouteCommand;
+import io.searchbox.cluster.reroute.RerouteMove;
+import io.searchbox.common.AbstractIntegrationTest;
+import io.searchbox.core.Cat;
+import io.searchbox.core.CatResult;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set ;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 3)
+public class RerouteIntegrationTest extends AbstractIntegrationTest {
+
+    static final String INDEX = "reroute";
+
+    @Before
+    public void beforeTest() throws IOException {
+        createIndex(INDEX);
+        ensureSearchable(INDEX);
+        setAllocationDisabled(true);
+    }
+
+    @After
+    public void afterTest() throws IOException {
+        setAllocationDisabled(false);
+    }
+
+    @Test
+    public void move() throws IOException, InterruptedException {
+        int shardToReroute = 0;
+
+        String fromNode = getNodeOfPrimaryShard(INDEX, shardToReroute);
+        String toNode = getAvailableNodeForShard(INDEX, shardToReroute);
+
+        RerouteMove rerouteMove = new RerouteMove(INDEX, shardToReroute, fromNode, toNode);
+        JestResult result = client.execute(new Reroute.Builder(rerouteMove).build());
+        assertTrue(result.getErrorMessage(), result.isSucceeded());
+
+        // Wait for shard to be moved
+        Thread.sleep(1000);
```

Comment:
can fail the test if took more than 1 second...
maybe do wait until or something...